src/parcels/_core/utils/sgrid.py

```diff
@@ -378,3 +392,95 @@ def parse_sgrid(ds: xr.Dataset):
         xgcm_coords[axis] = {"center": dim_dim_padding.dim2, xgcm_position: dim_dim_padding.dim1}
 
     return (ds, {"coords": xgcm_coords})
+
+
+def rename_dims(ds: xr.Dataset, dims_dict: dict[str, str]) -> xr.Dataset:
+    grid_da = get_grid_topology(ds)
+    if grid_da is None:
+        raise ValueError(
+            "No variable found in dataset with 'cf_role' attribute set to 'grid_topology'. Is this an SGrid dataset?"
```

Comment:
Not sure the question at the end of this Error message is useful. There's not way for the user to answer "Yes", or even if they do, that won't change the error. Either remove the last part, or add an actionable hint?